src/core/exec-invoke.c

```diff
@@ -2280,6 +2280,56 @@ static int create_many_symlinks(const char *root, const char *source, char **sym
         return 0;
 }
 
+static int is_idmapping_supported(const char *path) {
+        _cleanup_close_ int mount_fd = -EBADF, userns_fd = -EBADF, dir_fd = -EBADF;
+        _cleanup_free_ char *uid_map = NULL, *gid_map = NULL;
+        int r;
+
```

Comment:
In order to avoid calling this again and again for every directory, please add a `static int support_cached = -1` or so, and if it is set to 0 or 1 return it immediately as the first thing, otherwise proceed and set it below, so that it happens only once. If the kernel doesn't support this, it won't change across multiple runs.

Reply:
The check depends on the input path though, i.e. checking if it's a directory, mountpoint, underlying FS, all of which may not be necessarily the across all directories?, so the syscall may work/be supported for some exec dirs but not others

Reply:
Ah yes I forgot that it's implemented filesystem-by-filesystem, and not globally in the kernel, never mind then, ignore this